playground/nextjs-app-router/lib/url-params.ts

```diff
@@ -0,0 +1,68 @@
+import { OnchainKitComponent } from '@/types/onchainkit';
+
+const commonOptions = [
+  'activeComponent',
+  'componentMode',
+  'componentTheme',
+  'walletType',
+];
+
+const URL_PARAM_MAPPING: Partial<Record<OnchainKitComponent, string[]>> = {
+  [OnchainKitComponent.Checkout]: [
+    'chargeId',
+    'productId',
+    'checkoutTypes',
+    'isSponsored',
+    'checkoutOptions',
+  ],
+  [OnchainKitComponent.Transaction]: ['chainId', 'transactionType', 'calls'],
+  [OnchainKitComponent.TransactionDefault]: [
+    'chainId',
+    'transactionType',
+    'calls',
+  ],
+  [OnchainKitComponent.Swap]: ['chainId', 'defaultMaxSlippage'],
+  [OnchainKitComponent.SwapDefault]: ['chainId', 'defaultMaxSlippage'],
+  [OnchainKitComponent.NFTCard]: ['chainId', 'nftToken'],
+  [OnchainKitComponent.NFTCardDefault]: ['chainId', 'nftToken'],
+  [OnchainKitComponent.NFTMintCard]: ['chainId', 'nftToken'],
+  [OnchainKitComponent.NFTMintCardDefault]: ['chainId', 'nftToken'],
+};
+
+export function getShareableUrl(activeComponent?: OnchainKitComponent) {
+  if (!activeComponent) {
+    return window.location.origin;
+  }
+
+  const relevantParams = getComponentQueryParams(activeComponent);
+  const params = new URLSearchParams();
+
+  for (const param of [...relevantParams, ...commonOptions]) {
+    const value = localStorage.getItem(param);
+    if (value) {
+      params.set(param, value);
+    }
+  }
+  return `${window.location.origin}?${params.toString()}`;
```

Comment:
thoughts on only setting params that are not the default to reduce the length of the url?

Reply:
done! 👍🏻